run_project_tests.py

```diff
@@ -125,14 +125,14 @@ def platform_fix_name(fname, compiler, env):
         canonical_compiler = compiler
 
     if '?lib' in fname:
-        if mesonlib.for_windows(env.is_cross_build(), env) and canonical_compiler == 'msvc':
+        if env.machines.host.is_windows() and canonical_compiler == 'msvc':
             fname = re.sub(r'lib/\?lib(.*)\.', r'bin/\1.', fname)
             fname = re.sub(r'/\?lib/', r'/bin/', fname)
-        elif mesonlib.for_windows(env.is_cross_build(), env):
+        elif env.machines.host.is_windows():
             fname = re.sub(r'lib/\?lib(.*)\.', r'bin/lib\1.', fname)
             fname = re.sub(r'\?lib(.*)\.dll$', r'lib\1.dll', fname)
             fname = re.sub(r'/\?lib/', r'/bin/', fname)
-        elif mesonlib.for_cygwin(env.is_cross_build(), env):
+        elif env.machines.host.is_cygwin():
```

Comment:
tangent: looking at this I wonder if it would be better to have an `OperatingSystem(enum.Enum)` and just do `if env.machines.host.os is OperatingSystem.windows`? This would make checking for multiple oses much nicer:
`if env.machines.host.os in [OperatingSystem.windows, OperatingSystem.cygwin]`

not something to resolve in this series.

Reply:
Definitely more structure, but I feel like making windows and cygwin disjoint is also wrong. We'll decide later what's good.

Reply:
From what I've seen windows and cygwin do overlap, but in the same way that Linux and FreeBSD overlap, it's not the 90% that's the same that matters, it's the 10% that's different. It might be nice for our enum to define a `windows_like()` or `unix_like()` property so you could do something like:

```python
if env.machines.host.os.windows_like:
    ....
elif env.machines.host.os.unix_like:
    ...
```

To cover the 90% cases.

Reply:
I'll write this up and send a MR and we can discuss this further there.

Reply:
Yeah that sounds perfect. It's sort of like what we did in my distro in https://github.com/NixOS/nixpkgs/blob/master/lib/systems/inspect.nix#L42-L44, which is inspired by what LLVM does http://llvm.org/doxygen/Triple_8cpp_source.html#l00944 llvm.org/doxygen/Triple_8h_source.html#l00220 with `windows-gnu` vs `windows-cygwin` vs `windows-msvc`.